src/groups/mqb/mqbstat/mqbstat_queuestats.cpp

```diff
@@ -544,6 +544,15 @@ void QueueStatsDomain::onEvent(EventType::Enum type, bsls::Types::Int64 value)
     case EventType::e_ADD_MESSAGE: {
         d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_BYTES, value);
         d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_MESSAGES, 1);
+        if (!d_subContextsHolder.empty()) {
+            bsl::list<StatSubContextMp>::iterator it =
+                d_subContextsHolder.begin();
+            while (it != d_subContextsHolder.end()) {
```

Comment:
We got rid of the loop in the `put` method and here it is back again.
At the moment, I do not see a good way to utilize `VirtualStorageCatalog::numMessages` so let's keep the loop.
Provided we won't maintain statistics for all Dynamic Apps.

Reply:
Check for empty `d_subContextsHolder` should be fast if we don't have subcontexts for dynamic apps.
We might utilize `numMessages` on the time when we do snapshot, but this change is more complex, since we don't have a good way to ensure that each and every virtual storage is alive, and this change is intrusive.
I would rather merge this PR for now, since it helps to test and compare queue metrics. We can think of this in the future, when we have `prof` outputs for the scenario in question.

Reply:
Actually, I would also like to move metrics update code to the header and mark it `inline`, with hope that it will have less impact on performance. We have inline functions defined for `StatContext`, but we break the inline chain here when we define in `.cpp`. I would also like to do this change after we merge metrics update.